waiter/src/waiter/auth/composite.clj

```diff
@@ -0,0 +1,51 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns waiter.auth.composite
+  (:require [plumbing.core :as pc]
+            [waiter.auth.authentication :as auth]
+            [waiter.auth.saml :as saml]
+            [waiter.util.utils :as utils]))
+
+(defrecord CompositeAuthenticator [authenticators saml-acs-handler-fn saml-auth-redirect-handler-fn]
```

Comment:
The `saml-*` functions should not be part of the `CompositeAuthenticator `

Reply:
done